src/runtime/generators.js

```diff
@@ -147,7 +180,15 @@
         ctx.GState = ST_EXECUTING;
         ctx.action = action;
         ctx.sent = x;
-        var value = moveNext(ctx);
+        try {
+          var value = moveNext(ctx);
```

Comment:
move var out of try. Even though var are hoisted it is a good practice to not depend on it.